### ISSUE TITLE
feat: common dubbo-proxy plugin

### ISSUE THREAD

**Issue by mayunSaicmotor** [apache/apisix#10848]
### Description

Dear，
It seems it is not easy to use the dubbo proxy plugin in real project. It has too much restriction. It's not very common solution.
Do you have plan to  develop a common dubbo proxy plugin in future?

**Comment by sheharyaar** [comment]
@mayunSaicmotor , as we discussed in the previous issue #10648 , I would summarise the issues we faced :

- mod_dubbo used by APISIX, does not any datatype other than `Map<String, Object>` for both as return type and arguments.
- mod_dubbo uses hessian2 as the only deserialization protocol
- data passed through headers work out of the box, but for data in the body, it requires further deserialization using hessian2 decoder.

These are the issues that need to be worked upon to make dubbo-proxy better.

**Comment by Vacant2333** [comment]
Agree!  how do u think about this feat @shreemaan-abhishek @monkeyDluffy6017

**Comment by sheharyaar** [comment]
@shreemaan-abhishek , can you please assign me this

**Comment by sheharyaar** [comment]
Hi @mayunSaicmotor , can you please checkout this : #10703 , usage example can be found under `t/plugin/http-dubbo.t`

**Comment by monkeyDluffy6017** [comment]
LGTM
